docs/core/howto/listings/trial/calculus/client_1.py

```diff
@@ -18,7 +18,7 @@ def lineReceived(self, line):
     def _sendOperation(self, op, a, b):
         d = defer.Deferred()
         self.results.append(d)
-        line = "%s %d %d" % (op, a, b)
+        line = ("%s %d %d" % (op, a, b)).encode('utf-8')
```

Comment:
Additionally, for consistent behavior on Python 2 and 3, any place you are doing an encode on a string, please prefix with *u*, so this will be:

```
u"{} {} {}".format(op, a, b)
```